Fix a race condition where a thread could be put on the thread list,
then canceled before we set up cleanup push protection.  This is
probably a temporary fix.

diff -Naurb libmonitor-0.x-r116.orig/src/pthread.c libmonitor-0.x-r116/src/pthread.c
--- libmonitor-0.x-r116.orig/src/pthread.c	2011-04-28 16:54:40.343473758 -0500
+++ libmonitor-0.x-r116/src/pthread.c	2011-04-28 16:56:12.703655948 -0500
@@ -530,7 +530,7 @@
 	    if (tn->tn_appl_started && !tn->tn_fini_started)
 		(*real_pthread_kill)(tn->tn_self, MONITOR_EXIT_CLEANUP_SIGNAL);
 
-	    if (! tn->tn_fini_done)
+	    if (tn->tn_appl_started && !tn->tn_fini_done)
 		num_unfinished++;
 	}
 	MONITOR_DEBUG("waiting on %d threads to finish\n", num_unfinished);
